starfish/test/full_pipelines/api/test_dartfish.py

```diff
@@ -0,0 +1,156 @@
+import numpy as np
+import pandas as pd
+
+from starfish import Codebook, Experiment
+from starfish.image._filter.scale_by_percentile import ScaleByPercentile
+from starfish.image._filter.zero_by_channel_magnitude import ZeroByChannelMagnitude
+from starfish.spots._detector.pixel_spot_detector import PixelSpotDetector
+from starfish.types import Features
+
+
+def test_dartfish_pipeline_cropped_data():
+
+    # load the experiment
+    experiment_json = (
+        'https://dmf0bdeheu4zf.cloudfront.net/'
+        'TEST/20180821/DARTFISH_TEST/fov_001/experiment.json'
+    )
+    experiment = Experiment.from_json(experiment_json)
+
+    primary_image = experiment.image
+    primary_image._data = primary_image._data.astype(float)
+
+    expected_primary_image = np.array(
```

Comment:
I'm also a little anxious about this kind of test but I don't have a better idea. Right now I'd rather see these start breaking, realize I've made a mistake, and come back and rip them out later. 

@dganguli and I worked through some pretty shitty bugs last time I was in town that these tests would catch.

Reply:
I think for most types of filters, I think there is a strong expectation that the output should always be the same for a constant input. However, things that use optimization (e.g., some deconvolution algorithms or some spot detectors) may not converge to precisely the same result. 

Perhaps there should be different criteria for different types of pipeline components (i.e., exactly equal or within some tolerance). Of course, this opens the question of how do we choose a tolerance...

Reply:
This is an extremely helpful comment. Some thoughts: 

(1) it's only filters for which I'm doing exact testing. 

(2) For the deconvolution problem, I will set the random seed so that it should converge the same way each time for the tests -- I wasn't aware it that richardson-lucy was random... embarrassing. 

(3) I'm hesitant to go the tolerance route in case of bias in the pipeline somewhere (slow numerical creep in one direction over many PRs).

Reply:
Are you referring to #397?

Reply:
@ttung if you mean this comment: 

> @dganguli and I worked through some pretty shitty bugs last time I was in town that these tests would catch.

Then yes, #397, but also #395. I'm sure I've spent at least 40 hours searching for needles in a haystack which these tests would help localize and prevent :)